Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasFunction_Source.jinja

```diff
@@ -0,0 +1,178 @@
+        {% for function in Library.findall('Function') %}
+        {
+{%- set hasbranch = CheckFunctionBranchResult(function) -%}
+{% if hasbranch|booleanTrue == true %}
+{% set branchfunction = function.attrib['Branch'] %}
+                AZ::BranchOnResultInfo branchResultInfo;
+{% if branchfunction is defined and branchfunction and not branchfunction == "Boolean" %}
+{% set sanitizedBranchFunctionName = macros.CleanName(branchfunction).replace('::', '_') %}
+                branchResultInfo.m_nonBooleanResultCheckName = "{{sanitizedNamespaceName}}_{{sanitizedBranchFunctionName}}";
+{% endif -%}
+{% set branchwithvalue = function.attrib['BranchWithValue'] %}
+{%- if branchwithvalue is defined and branchwithvalue.lower() == "true" %}
+                branchResultInfo.m_returnResultInBranches = true;
+{% endif -%}
+{{SetBranchExecution(function)}}
+{%- endif -%}
+
+
+            {% set functionName = macros.CleanName(function.attrib['Name']) %}
+            {% set sanitizedFunctionName = macros.CleanName(function.attrib['Name']).replace('::', '_') %}
+            {
+             behaviorContext->Method("{{sanitizedNamespaceName}}_{{sanitizedFunctionName}}", &{{className}}::{{functionName}},
+{{macros.GenerateFunctionMetaData(function)}})
+                    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+{% if categoryName %}
+                    ->Attribute(AZ::Script::Attributes::Category, "{{categoryName}}")
+{% endif %}
+{% if hasbranch|booleanTrue == true %}
+                    ->Attribute(AZ::ScriptCanvasAttributes::BranchOnResult, branchResultInfo)
+{% endif %};
+            }
+        }
+        {% endfor %}
+
+
+
+        {# Reflect Data Types to Behavior Context #}
+{%      if Library.tag == "Data" %}
+{%         set dataName = Library.attrib['Name'] %}
+{%         set namespaceName = '' %}
+{%         set sanitizedNamespaceName = '' %}
+{%         if Library.attrib['Namespace'] is defined and Library.attrib['Namespace'] %}
+{%             set namespaceName = Library.attrib['Namespace'] + '::' %}
+{%             set sanitizedNamespaceName = macros.CleanName(Library.attrib['Namespace'].replace('::', '_')) + '_' %}
+{%         endif %}
+            behaviorContext->Class<ContainerTypeReflection::BehaviorClassReflection<{{namespaceName}}{{dataName}}>>("ScriptCanvas_{{sanitizedNamespaceName}}{{dataName}}")
+                ->Attribute(AZ::Script::Attributes::ExcludeFrom, AZ::Script::Attributes::ExcludeFlags::All)
+                ->Attribute(AZ::Script::Attributes::Ignore, true)
+                ->Method("ScriptCanvas_ReflectVector", [](const AZStd::vector<{{namespaceName}}{{dataName}}>&) {})
+                ->Method("ScriptCanvas_ReflectStringMap", [](const AZStd::unordered_map<AZStd::string, {{namespaceName}}{{dataName}}>&) {})
+                ->Method("ScrtptCanvas_ReflectNumberMap", [](const AZStd::unordered_map<double, {{namespaceName}}{{dataName}}>&) {});
```

Comment:
I recommend using a type alias for the String and NumberMap in these function to make it a bit easier for reuse and readability.
```suggestion
            using VectorType = AZStd::vector<{{namespaceName}}{{dataName}}>;
            using StringMap = AZStd::unordered_map<AZStd::string, {{namespaceName}}{{dataName}}>;
            using NumberMap = AZStd::unordered_map<double, {{namespaceName}}{{dataName}}>
            behaviorContext->Class<ContainerTypeReflection::BehaviorClassReflection<{{namespaceName}}{{dataName}}>>("ScriptCanvas_{{sanitizedNamespaceName}}{{dataName}}")
                ->Attribute(AZ::Script::Attributes::ExcludeFrom, AZ::Script::Attributes::ExcludeFlags::All)
                ->Attribute(AZ::Script::Attributes::Ignore, true)
                ->Method("ScriptCanvas_ReflectVector", [](const VectorType&) {})
                ->Method("ScriptCanvas_ReflectStringMap", [](const StringMap&) {})
                ->Method("ScrtptCanvas_ReflectNumberMap", [](const DoubleMap&) {});
```